lemur/static/app/angular/destinations/destination/destination.js

```diff
@@ -52,19 +52,19 @@ angular.module('lemur')
           if (plugin.slug === $scope.destination.plugin.slug) {
             plugin.pluginOptions = $scope.destination.plugin.pluginOptions;
             $scope.destination.plugin = plugin;
-            _.each($scope.destination.plugin.pluginOptions, function (option) {
-              if (option.type === 'export-plugin') {
-                PluginService.getByType('export').then(function (plugins) {
-                  $scope.exportPlugins = plugins;
+            PluginService.getByType('export').then(function (plugins) {
+              $scope.exportPlugins = plugins;
```

Comment:
Order of these calls swapped. If a destination was edited and it did NOT use an export plugin yet (e.g. if "AWS" was selected), then selecting a new destination plugin that _did_ require an export plugin resulted in the export plugin selection list being empty, since we never loaded the possible plugins. By swapping, we _always_ load the export plugins (even if we don't need them yet), so that we have them ready if the user tries to change the plugin type.